src/App.jsx

```diff
@@ -1,12 +1,30 @@
-import { useState } from 'react'
-import './App.css'
-import movieData from './mockData'
+import { useState, useEffect } from 'react'
 import AllMovies from './components/AllMovies/AllMovies'
 import SingleMovie from './components/SingleMovie/SingleMovie'
 
 export default function App() {
-  const [movies, setMovies] = useState(movieData.movies)
+  const [movies, setMovies] = useState([])
   const [selectedSingleMovie, setSelectedSingleMovie] = useState(null)
+  const [error, setError] = useState(null)
```

Comment:
This is probably more of a dev-opinion rather than true best-practice but consider how it might be beneficial to use an empty state of the expected data type rather than using null here.  Personally, I appreciate an empty state because then its clear to me and other devs right off the bat that the selectedSingleMovie state will be an array, object, etc.   With null, I have no idea what shape that data will eventually take.